plugins/modules/ec2_asg_scheduled_action.py

```diff
@@ -0,0 +1,329 @@
+  returned: when I(state=present)
+  type: int
+  sample: 1
+'''
+
+try:
+    import botocore
+except ImportError:
+    pass  # caught by AnsibleAWSModule
+
+try:
+    from dateutil.parser import parse as timedate_parse
+    HAS_DATEUTIL = True
+except ImportError:
+    HAS_DATEUTIL = False
+
+from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import AWSRetry
+
+
+def format_request():
+    params = dict(
+        AutoScalingGroupName=module.params.get('autoscaling_group_name'),
+        ScheduledActionName=module.params.get('scheduled_action_name'),
+        Recurrence=module.params.get('recurrence')
+    )
+
+    # Some of these params are optional
+    if module.params.get('desired_capacity') is not None:
+        params['DesiredCapacity'] = module.params.get('desired_capacity')
+
+    if module.params.get('min_size') is not None:
+        params['MinSize'] = module.params.get('min_size')
+
+    if module.params.get('max_size') is not None:
+        params['MaxSize'] = module.params.get('max_size')
+
+    if module.params.get('time_zone') is not None:
+        params['TimeZone'] = module.params.get('time_zone')
+
+    if module.params.get('start_time') is not None:
+        params['StartTime'] = module.params.get('start_time')
+
+    if module.params.get('end_time') is not None:
+        params['EndTime'] = module.params.get('end_time')
+
+    return params
+
+
+def delete_scheduled_action(current_actions):
+    if current_actions == []:
+        return False
+
+    if module.check_mode:
+        return True
+
+    params = dict(
+        AutoScalingGroupName=module.params.get('autoscaling_group_name'),
+        ScheduledActionName=module.params.get('scheduled_action_name')
+    )
+
+    try:
+        client.delete_scheduled_action(aws_retry=True, **params)
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=str(e))
+
+    return True
+
+
+def get_scheduled_actions():
+    params = dict(
+        AutoScalingGroupName=module.params.get('autoscaling_group_name'),
+        ScheduledActionNames=[module.params.get('scheduled_action_name')]
+    )
+
+    try:
+        actions = client.describe_scheduled_actions(aws_retry=True, **params)
+    except botocore.exceptions.ClientError as e:
+        module.fail_json_aws(e)
+
+    current_actions = actions.get("ScheduledUpdateGroupActions")
+
+    return current_actions
+
+
+def put_scheduled_update_group_action(current_actions):
+    changed = False
+    changes = dict()
+    params = format_request()
+
+    if len(current_actions) < 1:
+        changed = True
+    else:
+        for k, v in params.items():
+            # To correctly detect changes convert the start_time & end_time to datetime object
+            if k == "StartTime":
+                v = timedate_parse(v)
+            elif k == "EndTime":
+                v = timedate_parse(v)
```

Comment:
There's no need to loop to find the parameters 

```suggestion
        # To correctly detect changes convert the start_time & end_time to datetime object
        if "StartTime" in params:
            params["StartTime"] = timedate_parse(params["StartTime"])
        if "EndTime" in params:
            params["EndTime"] = timedate_parse(params["EndTime"])
        for k, v in params.items():
```